dask_saturn/core.py

```diff
@@ -121,3 +130,33 @@ def close(self):
             raise ValueError(response.reason)
         for pc in self.periodic_callbacks.values():
             pc.stop()
+
+class ExpBackoff:
+    def __init__(self, wait_timeout=1200, min_sleep=5, max_sleep=60):
+        self.wait_timeout = wait_timeout
+        self.max_sleep = max_sleep
+        self.min_sleep = min_sleep
+        self.retries = 0
+    
+    def wait(self):
+        if self.retries == 0:
+            self.start_time = datetime.now()
+
+        # Check if timeout has been reached
+        time_delta = (datetime.now() - self.start_time).seconds
```

Comment:
I think this should be `.total_seconds()`, not `.seconds`. If the difference is more than a day, .seconds is going to give you a misleading answer.
```suggestion
        time_delta = (datetime.now() - self.start_time).total_seconds()
```

```
>>> d =  datetime(2020, 1, 1, 1, 1, 10) -  datetime(2020, 1, 1, 1, 1, 1)
>>> d.seconds
9
>>> d.total_seconds()
9.0
>>> 
>>> d =  datetime(2020, 1, 2, 1, 1, 10) -  datetime(2020, 1, 1, 1, 1, 1)
>>> 
>>> d.seconds
9
>>> 
>>> d.total_seconds()
86409.0
```

Reply:
good catch!